### ISSUE TITLE
with_items variable name template broken (worked in 1.9.4)

### ISSUE THREAD

**Issue by robinro** [ansible/ansible#13584]
##### Ansible Version:

ansible 2.0.0 (f41dd578b340b76aa706af9540fd8fab2fbb0eb6)
##### Ansible Configuration:

self-packaged version based on above revision
##### Environment:

openSuSE Leap 42.1
##### Summary:

Substituting variable names based on other variables does not work in ansible 2.0.0 but used to work in 1.9.4.

This should either be fixed or stated more prominently in the Changelog as it breaks existing playbooks.
##### Steps To Reproduce:

Example playbook:

```
vars:
  packages_install_42:
    - bash
    - zsh

tasks:
- name: install stuff
  zypper: name={{item}} state=present
  with_items: packages_install_{{ansible_distribution_major_version}}
```
##### Expected Results:

Variable packages_install_42 is used, the packages bash and zsh are installed.
##### Actual Results:

The task fails:
`(item=[u'packages_install_42']) => {"failed": true, "item": ["packages_install_42"], "msg": "No provider of 'packages_install_42' found.\n", "rc": 104}`

**Comment by bcoca** [comment]
I don't think that is supposed to work, this is how it should be written:
`with_items: "{{ hostvars[inventory_hostname]['packages_install_' + ansible_distribution_major_version] }}"`

**Comment by robinro** [comment]
**Edit**: Ignore this comment, the case there is different (not a variable)

~~It is/was suppossed to work. At least it is in the docs: 
http://docs.ansible.com/ansible/playbooks_loops.html#finding-first-matched-files~~

**Comment by rektide** [comment]
I too used this old syntax. I'm fine with @bcoca's suggestion, but I can't get it working. Here's a dumb test playbook, which shows that variables in var: are not found in either hostvars[inventory_hostname] or vars;

```
---
- hosts: all
  gather_facts: False
  vars:
    LIST:
    - one
    - two
  tasks:
   - debug:
     args:
       msg: "{{ hostvars[inventory_hostname] }}"
   - debug:
     args:
       msg: "{{ vars }}"
   - debug:
     args:
       msg: "{{ hostvars[inventory_hostname]['LIST'] }}"
```

(Same result without gather_facts, it just makes output easier to read.)

I adopted this syntax because of #5000, which is where I originally found that vars: were not found in hostvars or vars. I'm fine with the syntax going away, but now as before in #5000, there's no way to indirectly reference vars from vars:, as far as I can tell.

**Comment by bcoca** [comment]
@rektide a fix for your issue #13626

**Comment by robinro** [comment]
I don't have a strong opinion to keep the above-mentioned version working. Maybe this should make it into the changelog as it breaks existing playbooks.
@bcoca: if you don't plan to implement the old behavior and nobody else complains, this issue can be closed.

**Comment by Vladimir-csp** [comment]
Tried to upgrade today and stumbled upon this very issue.
I have package install/remove definitions feeded by `with_items: base_packages_{{ ansible_lsb.id }}_{{ ansible_lsb.codename }}` and like in almost every role, all broken.

**Comment by Vladimir-csp** [comment]
Or maybe there is some 'more proper' way to accomplish this task?

**Comment by bcoca** [comment]
the 'supported' var interpolation was using vars or hostvars (depending on where the var is)

```
with_items: "{{vars['base_packages_' + ansible_lsb.id  + '_' + ansible_lsb.codename] }}"
```

**Comment by Vladimir-csp** [comment]
It works, but 1.9 syntax is way simpler.

**Comment by bcoca** [comment]
Well, that was never intended to work, I had not seen it until today.

Many people tried the same in other options and we always pointed at the hostvars/vars way of doing it. This probably only works in with_ by a cascade of  bugs, many of which were fixed in 2.0.

**Comment by bcoca** [comment]
fixed via d6283129e13e1be5fa41b1440a0ab8016cd1b50c
